### ISSUE TITLE
Crash on iOS11 on devices with 1 GB RAM

### ISSUE THREAD

**Issue by lec00q** [cordova-plugin-camera-preview/cordova-plugin-camera-preview#393]
I experience repeatable crashes when taking a series of pictures using the camera preview. The issue is present on iPhone devices with 1GB of RAM, such as the iPhone 6 or iPhone 6 Plus, and it has arose after the introduction of iOS11.

Not sure but it can be the same or related to #200. 

I do take pictures normally:

```
window.CameraPreview.takePicture(this.photoSize, (imageFile) => {
    this.onPictureTaken(imageFile);
}, (error) => {
   console.log("error", error);
});
}, false);
```

`onPictureTaken` simply upload the photo to our server.

The crash happens after 5 to 10 pictures taken in rapid succession. ~I suspect that the OS is simply killing the app because of too high memory usage.~

Can you help me? Is there something that you need to debug the issue?
Thank you very much

**Comment by lec00q** [comment]
This is the log when debugging with XCode on iPhone 6:

```
2017-11-30 11:04:40.891146+0100 Snapfeet[280:9662] Done creating still image
2017-11-30 11:04:42.311040+0100 Snapfeet[280:9662] takePicture
2017-11-30 11:04:42.694112+0100 Snapfeet[280:9662] takePicture
2017-11-30 11:04:42.964057+0100 Snapfeet[280:9662] Done creating still image
2017-11-30 11:04:43.597882+0100 Snapfeet[280:9662] Done creating still image
2017-11-30 11:04:44.797540+0100 Snapfeet[280:9662] takePicture
2017-11-30 11:04:44.922928+0100 Snapfeet[280:9662] takePicture
2017-11-30 11:04:44.927216+0100 Snapfeet[280:9662] Done creating still image
2017-11-30 11:04:45.381102+0100 Snapfeet[280:9662] Done creating still image
2017-11-30 11:04:46.259087+0100 Snapfeet[280:9662] takePicture
2017-11-30 11:04:46.368569+0100 Snapfeet[280:9662] Done creating still image
2017-11-30 11:04:48.100273+0100 Snapfeet[280:9662] Got memory pressure notification (non-critical)
2017-11-30 11:04:48.175179+0100 Snapfeet[280:9662] takePicture
2017-11-30 11:04:48.301876+0100 Snapfeet[280:9662] Done creating still image
2017-11-30 11:04:48.958193+0100 Snapfeet[280:9662] takePicture
2017-11-30 11:04:48.961135+0100 Snapfeet[280:9662] takePicture
2017-11-30 11:04:49.147013+0100 Snapfeet[280:9662] Done creating still image
2017-11-30 11:04:49.717817+0100 Snapfeet[280:9662] Done creating still image
2017-11-30 11:04:50.829529+0100 Snapfeet[280:9662] THREAD WARNING: ['File'] took '27.026855' ms. Plugin should use a background thread.
2017-11-30 11:04:50.832959+0100 Snapfeet[280:9662] takePicture
2017-11-30 11:04:50.887914+0100 Snapfeet[280:9662] THREAD WARNING: ['CameraPreview'] took '54.916016' ms. Plugin should use a background thread.
2017-11-30 11:04:50.914619+0100 Snapfeet[280:9662] takePicture
2017-11-30 11:04:51.438745+0100 Snapfeet[280:9662] takePicture
2017-11-30 11:04:51.442116+0100 Snapfeet[280:9662] takePicture
2017-11-30 11:04:51.574610+0100 Snapfeet[280:9662] takePicture
2017-11-30 11:04:52.010158+0100 Snapfeet[280:9662] Done creating still image
2017-11-30 11:04:52.597196+0100 Snapfeet[280:9662] System is no longer under (non-critical) memory pressure.
2017-11-30 11:04:52.602364+0100 Snapfeet[280:9662] Done creating still image
2017-11-30 11:04:53.972517+0100 Snapfeet[280:9662] Done creating still image
2017-11-30 11:04:55.233331+0100 Snapfeet[280:9662] Done creating still image
2017-11-30 11:04:55.787763+0100 Snapfeet[280:9662] Done creating still image
```

But I cannot attach to the process because the app simply crashes and the connection between XCode and the phone drops.

**Comment by lec00q** [comment]
**Update**:

Debugging on Xcode is not helpful at all because the app crashes without any stack trace.
I am not very expert in those stuffs but I am quite sure that the app crashes on this line:

```
CGImageRef finalImage = [self.cameraRenderController.ciContext createCGImage:finalCImage fromRect:finalCImage.extent];
```

in `CameraPreview.m`. I can confirm that the crashes happen when trying to take pictures in a rapid succession, tapping multiple times on the screen and only after a few pictures taken. 

Any help would be really appreciated!

**Comment by ngocnguyenvan** [comment]
I also have the same case, even it was in iOS 10 with iphone 6 and iphone 6 plus. please help

**Comment by lucas-entronic** [comment]
Same, on iPhone 6 Plus and iPhone 7 with iOS 11.

**Comment by PicklesAU** [comment]
Same, on iPhone 8
i take picture 1 at a time, (takepicture, then reopen the page take another pic)
but this still crash after taking the 11th pic.

**Comment by ngt14** [comment]
Same issue on iPad Mini

**Comment by ryaa** [comment]
This problem could be related to https://github.com/cordova-plugin-camera-preview/cordova-plugin-camera-preview/issues/604

**Comment by westonganger** [comment]
Closed via #604 and #605
